auth/migrations/0000_init.sql

```diff
@@ -1,5 +1,7 @@
+CREATE TYPE tier AS ENUM ('basic', 'pendingpaymentpro', 'pro', 'team', 'admin', 'deployer');
```

Comment:
I would suggest to keep using `TEXT` for now. in case we need to revert. We can do the other migrations and use more of Postgres feature later on.

Reply:
What would it look like in the end? 

I had trouble in making sqlx-postgres work with Rust enums like our `AccountTier`. What's a bit off-putting is this part of the docs [1]. It requires the creation of a postgres type as well (so double tracking) when trying to work seamlessly with Rust enums <-> Postgres type, and be able to get the data from postgres coercing it to `AccountTier` Rust enum. I haven't verified specifically if sending data works with `TEXT` type (I believe it should), but coercing it to the `AccountTier` enum is subject to type errors.

If we want to work just with `TEXT`, then we need to get it from Postgres and then convert it to `AccountTier`, which should be fine too. Will try to look more into a seamless conversion between the two, but happy to hear other suggestions too.

[1] https://docs.rs/sqlx/latest/sqlx/postgres/types/index.html#enumerations

Reply:
It would look exactly like your last paragraph :) 

We currently have that issue with SQLite as well so it's fine to have it for the moment since it will make the migration easier.

Reply:
Not sure what issue we have with SQLite currently. We don't need a new SQL type to exist for the conversion, or `From` impls. However, I don't mind using a `From` to avoid double tracking (if it works), at the expense of not using the recommendation from the docs.

I also think the migration concern is important, so will try to make the adjustments and ping afterward.

Reply:
Done.

Reply:
I meant the issue that we don't have any assurance that what's in the database matches the enum when using SQLite, since it's a text field.

Reply:
Ah yes, indeed.